libraries/rtos/rtx/TARGET_CORTEX_M/RTX_CM_lib.h

```diff
@@ -501,75 +553,20 @@ __asm void __rt_entry (void) {
 
 #elif defined (__GNUC__)
 
-#ifdef __CS3__
-
-/* CS3 start_c routine.
- *
- * Copyright (c) 2006, 2007 CodeSourcery Inc
- *
- * The authors hereby grant permission to use, copy, modify, distribute,
- * and license this software and its documentation for any purpose, provided
- * that existing copyright notices are retained in all copies and that this
- * notice is included verbatim in any distributions. No written agreement,
- * license, or royalty fee is required for any of the authorized uses.
- * Modifications to this software may be copyrighted by their authors
- * and need not follow the licensing terms described here, provided that
- * the new terms are clearly indicated on the first page of each file where
- * they apply.
- */
-
-#include "cs3.h"
-
+extern void __libc_fini_array(void);
 extern void __libc_init_array (void);
+extern int main(int argc, char **argv);
 
-__attribute ((noreturn)) void __cs3_start_c (void){
-  unsigned regions = __cs3_region_num;
-  const struct __cs3_region *rptr = __cs3_regions;
-
-  /* Initialize memory */
-  for (regions = __cs3_region_num, rptr = __cs3_regions; regions--; rptr++) {
-    long long *src = (long long *)rptr->init;
-    long long *dst = (long long *)rptr->data;
-    unsigned limit = rptr->init_size;
-    unsigned count;
-
-    if (src != dst)
-      for (count = 0; count != limit; count += sizeof (long long))
-        *dst++ = *src++;
-    else
-      dst = (long long *)((char *)dst + limit);
-    limit = rptr->zero_size;
-    for (count = 0; count != limit; count += sizeof (long long))
-      *dst++ = 0;
-  }
-
-  /* Run initializers.  */
-  __libc_init_array ();
-
-  osKernelInitialize();
-#ifdef __MBED_CMSIS_RTOS_CM
-  set_main_stack();
-#endif
-  osThreadCreate(&os_thread_def_main, NULL);
-  osKernelStart();
-  for (;;);
+void pre_main(void) {
+    atexit(__libc_fini_array);
+    __libc_init_array();
+    main(0, NULL);
```

Comment:
@c1728p9 Leaving it as is, is ok with me. 

When I build non-RTOS programs with the mbed SDK, I don't even call atexit() before calling main (the version of newlib I was using at the time I wrote my startup code would pull in malloc() for atexit(), I don't know if it still does).  That means exit() doesn't even call the destructors in my non-RTOS programs anyway. I do call exit() in my shutdown code but that is only because I like the semihost API that the mbed SDK ends up calling during exit (since it drops me into the debugger) and the blinking LEDs if an error is returned from main().